qiskit/algorithms/factorizers/shor.py

```diff
@@ -375,9 +392,48 @@ def _calculate_continued_fraction(b: array.array) -> int:
         logger.debug("Numerator:%s \t\t Denominator: %s.", frac.numerator, frac.denominator)
         return frac.denominator
 
-    def _run(self) -> AlgorithmResult:
-        if not self._ret['factors']:
-            logger.debug('Running with N=%s and a=%s.', self._N, self._a)
+    def factor(self,
+               N: int = 15,
```

Comment:
I find it very strange to give a default problem to an algorithm. This is to facilitate running the algorithm for the user, so couldn't we instead give these values as examples in the class docstring? Maybe we can even check if `N` or `a` are `None` and if they are raise a warning proposing these default values?

Reply:
agree, lets only give the default for `a=2` but not for `N`.

Reply:
fixed